src/components/scheduling/goals/SchedulingGoals.svelte

```diff
@@ -20,161 +18,164 @@
 
-  let filteredGoals: SchedulingGoal[] = [];
+  const baseColumnDefs: DataGridColumnDef<SchedulingGoalMetadata>[] = [
+    {
+      field: 'id',
+      filter: 'number',
+      headerName: 'ID',
+      resizable: true,
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 60,
+    },
+    { field: 'name', filter: 'text', headerName: 'Name', minWidth: 80, resizable: true, sortable: true },
+    {
+      field: 'owner',
+      filter: 'string',
+      headerName: 'Owner',
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 80,
+    },
+    {
+      field: 'updated_by',
+      filter: 'string',
+      headerName: 'Updated By',
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      width: 120,
+    },
+    {
+      field: 'versions',
+      filter: 'string',
+      headerName: 'Latest Version',
+      sortable: true,
+      suppressAutoSize: true,
+      suppressSizeToFit: true,
+      valueGetter: (params: ValueGetterParams<SchedulingGoalMetadata>) => {
+        return params?.data?.versions[params?.data?.versions.length - 1].revision;
+      },
+      width: 125,
+    },
+    {
+      autoHeight: true,
+      cellRenderer: tagsCellRenderer,
+      field: 'tags',
+      filter: 'text',
+      filterValueGetter: tagsFilterValueGetter,
+      headerName: 'Tags',
+      resizable: true,
+      sortable: false,
+      width: 220,
+      wrapText: true,
+    },
+  ];
+
+  const permissionError = 'You do not have permission to create a constraint.';
+
+  let columnDefs: DataGridColumnDef[] = baseColumnDefs;
+
+  let filteredGoals: SchedulingGoalMetadata[] = [];
   let filterText: string = '';
+  let hasPermission: boolean = false;
 
   $: if (schedulingGoals && $tagsMap) {
-    filteredGoals = schedulingGoals
-      .filter(goal => {
-        const filterTextLowerCase = filterText.toLowerCase();
-        const includesId = `${goal.id}`.includes(filterTextLowerCase);
-        const includesName = goal.name.toLocaleLowerCase().includes(filterTextLowerCase);
-        return includesId || includesName;
-      })
-      .map(goal => {
-        const matchingTags: { tag: Tag }[] = [];
-        goal.tags.forEach(({ tag_id }) => {
-          const matchingTag = $tagsMap[tag_id];
-          if (matchingTag) {
-            matchingTags.push({ tag: matchingTag });
-          }
-        });
-        const goalWithTags: SchedulingGoal = { ...goal, tags: matchingTags };
-        return goalWithTags;
-      });
+    filteredGoals = schedulingGoals.filter(goal => {
+      const filterTextLowerCase = filterText.toLowerCase();
+      const includesId = `${goal.id}`.includes(filterTextLowerCase);
+      const includesName = goal.name.toLocaleLowerCase().includes(filterTextLowerCase);
+      return includesId || includesName;
+    });
   }
+  $: hasPermission = featurePermissions.schedulingGoals.canCreate(user);
```

Comment:
The permission here is `CREATE_SCHEDULING_GOAL` which maps to `insert_scheduling_goal_one` which doesn't look like it exists, at least in my user's list of permissions, which results in the `new` button for goals and conditions being disabled when in role = `user`

Reply:
Though maybe my setup is not quite right, seeing `Error: field 'scheduling_specification' not found in type: 'plan'` again when trying to load a plan (though I can save scheduling goals...) so maybe I need to reset my dev instance. Though weirdly I already did do this so not sure why I keep running into this one.

Reply:
Thank you for catching that! This is exactly why I wanted the refactor haha.